### PR TITLE
[Topic recognition] brain MRI super-resolution network

### PR DESCRIPTION
**Pull request by Ericliol**
There are two tasks in the pull request, one is an easy project, and the other one this the normal difficulty one (brain MRI super-resolution network), please make the normal one. 
The model used in the task is called ESPCN (Efficient Sub-Pixel CNN), which reconstructs a high-resolution version of an image given a low-resolution version. It involves using convolutional neural networks (CNNs) and specifically a sub-pixel convolution layer to upscale low-resolution images to a higher resolution.

### CODE REVIEW

**Comment by uqmarlonbran** [comment]
**This is an initial inspection - no action is required yet**

**Difficulty:** Normal

**General Comments:** Code looks ok, however README or repository in general don't provide any training/test examples. No figures included at all.

**[Recognition Problem]**
"modules.py":

- Keras implementation similar to the Keras example.

"dataset.py":

- Uses image_dataset_from_directory to create an image dataset, then stores to global variables.
- Just has a training and "validation" set, but validation is from the test set (no actual validation set).

"train.py":

- Stores training and test losses.
- Implements callbacks.

"predict.py":

- Computes the PSNR for 10 images in the test set.
- Loads weights correctly.

"README.MD":

- README contains brief description of the code submitted, no visual aids such as plots or reconstruction examples.
- Supposed to have another folder called "test_results" with images but they haven't been included in the commit.

**[Commit Log]**
Commit log looks ok.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting minimal -1
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages sometimes -1
Progressive commits used
## Documentation
ReadMe OK, no plots, no usage, no dependencies -2
Model/technical explanation no architecture -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove pyc files, restore repo READMEs, rename your folder with unique name -2
Request Description minimal -1

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
